In [1]:
import numpy as np

# Part A

## Part i

In equilibrium we have $\sum F=0$ and hence $F_{grav}+F_{internal}+F_{external}=0$. Note that

$$F_{grav}=-mg$$
$$F_{internal}=P_{internal}A$$
$$F_{external}=-P_{ext}A$$

This yields $P_{int}=P_{ext}+\frac{mg}{A}$. We know that internal pressure is given by

$$P_{int}=\frac{\rho R T}{m_d}=\frac{MRT}{Vm_d}=\frac{MRT}{Ahm_d} $$

Isolating for $M$ yields

$$M=\frac{Ahm_d}{RT}\left[ P_{ext}+\frac{mg}{A}\right] $$

In [2]:
def to_K(temp):
    return temp+273.15
def to_C(temp):
     return temp-273.15

A=0.01; h=0.1
R=8.314; g=9.81
md = 29/1000
m = 2
T = to_K(20)
Pext = 1000*100

M = (A*h*md)/(R*T)*(Pext+m*g/A)

In [3]:
print('The Mass of internal air is {:.3f}g'.format(M*1000))

The Mass of internal air is 1.213g


## Part ii

The only difference now is that $T$ is changed. We can rearrange the equation above to get

$$h=\frac{MRT}{Am_d}\left[P_{ext}+\frac{mg}{A} \right]^{-1} $$

In [4]:
T = to_K(100)

h = M*R*T/(A*md)*(Pext+m*g/A)**(-1)

In [5]:
print('The height of the piston is {:.3f}cm'.format(h*100))

The height of the piston is 12.729cm


## Part iii

We use the same expression as before, only this time we replace the mass of the cylinder with 102kg.

In [6]:
m = 102
T  = to_K(20)

h = M*R*T/(A*md)*(Pext+m*g/A)**(-1)

In [7]:
print('The height of the piston is {:.3f}cm'.format(h*100))

The height of the piston is 5.097cm


## Part iv

The height needs to be 8.097cm. Solving for temperature...

In [8]:
h = h+3/100

T = A*h*md/(R*M)*(Pext+m*g/A)

In [9]:
print(r'The temperature in the piston needs to be {:.3f} degrees C'.format(to_C(T)))

The temperature in the piston needs to be 192.559 degrees C


# Part b

## Part i

In this scenario the vapour is saturated so its vapour pressure is given by 2.3.11. The total pressure is determined by setting the sum of the forces on the piston equal to zero. We know that

$$e=e_S(T)=e_0 e^{\frac{m_v L}{R}\left(\frac{1}{T_0}-\frac{1}{T} \right)}$$

$$m=\frac{m_d}{1+0.61q} $$

It follows that

$$q=\frac{m_ve}{mp}=\frac{m_v (e_S(T))(1+0.61q)}{m_dp} $$

and isolating for $q$ yields

$$q=\frac{m_ve_s(T)}{m_dp-0.61m_ve_s(T)} $$

In [10]:
mv = 18/1000
md = 29/1000
L = 2.45*10**6
e0 = 611
T0 = 273
T = to_K(20)
m = 2

p  = Pext+m*g/A

def es(T):
    return e0*np.exp(mv*L/R*(1/T0-1/T))

q = mv*es(T)/(md*p-0.61*mv*es(T))

In [11]:
print('The specific heat is {:.4f}'.format(q))

The specific heat is 0.0143


## Part ii

Since $p=MRT/Vm$ we have $M=pVm/RT$ and hence

$$M=\frac{pV}{RT}\left(\frac{m_d}{1+0.61q}\right) $$

In [12]:
M = p*A*h/(R*T)*(md/(1+0.61*q))

In [13]:
Mv = q*M
Md = M-q*M

print('The total mass in the bucket is {:.3f}g'.format(M*1000))
print('The mass of water vapour is thus {:.3f}g'.format(Mv*1000))
print('The mass of dry air is thus {:.3f}g'.format(Md*1000))

The total mass in the bucket is 0.974g
The mass of water vapour is thus 0.014g
The mass of dry air is thus 0.960g


## Part iii

The only thing that changes now is the internal pressure.

In [14]:
m = 102
p  = Pext+m*g/A

q = mv*es(T)/(md*p-0.61*mv*es(T))

In [15]:
print('The specific heat is {:.5f}'.format(q))

The specific heat is 0.00724


This changes the total mass in the chamber; we have

$$M_v = q(M_d+M_v) \implies M_v=\frac{q}{1-q}M_d$$

In [16]:
Mv = q/(1-q)*Md
M=Mv+Md

print('The new mass of water vapour is {:.5f}g'.format(Mv*1000))
print('Thus the total mass in the chamber is is {:.5f}g'.format(M*1000))

The new mass of water vapour is 0.00700g
Thus the total mass in the chamber is is 0.96691g


The height of the cylinder is obtained using

$$p=\frac{MRT}{V}=\frac{MRT}{Ahm} \implies h=\frac{MRT}{AP}\frac{1+0.61q}{m_d}$$

In [17]:
h = M*R*T/(A*p)*(1+0.61*q)/md
print('The height of the cylinder is thus {:.3f}cm'.format(h*100))

The height of the cylinder is thus 4.080cm


## Part iv

Now $e_s(T)$ changes since we are changing the temperature.

In [18]:
T=to_K(100)
q = mv*es(T)/(md*p-0.61*mv*es(T))

In [19]:
print('The new value of the specific heat is {:.3f}'.format(q))

The new value of the specific heat is 0.443


Now we compute the total mass:

In [20]:
Mv = q/(1-q)*Md
M=Mv+Md

In [21]:
print('The new total mass in the container is {:.3f}g'.format(M*1000))

The new total mass in the container is 1.723g


Finally we calculate the height:

In [22]:
h = M*R*T/(A*p)*(1+0.61*q)/md
print('The height of the cylinder is thus {:.3f}cm'.format(h*100))

The height of the cylinder is thus 11.704cm


# Part C

In part a) we showed that the temperature inside the piston needed to be $192^\circ C$ to raise the piston (with the $100$kg mass on top) $8.097$cm. In part b) we showed that the temperature inside the piston (with water vapour) only needed to be $100^\circ C$ to raise the pison $11.7$cm. Clearly the piston with water vapour is more effective at lifting than the piston without water vapour.

If one wants to raise the piston most efficiently, one needs to maximize the internal pressure of the piston. Now since

$$P=\frac{NRT}{V} $$

and $V$ is constant (during a short interval of time), one can maximize the lifting force by increasing $N$ and $T$. In the case of dry air, $N$ is constant (mass of dry air is constant) so the only way one increase the internal pressure of the system is raising the temperature.In the case of moist air, raising the temperature has the additional effect of increasing the number of gaseous water molecules (provided there is an aqueous layer of water at the bottom of the piston). This additional effect raises the pressure even more, and thus the water-filled piston is more effective at lifting the mass than the dry air piston.